src/encoded/static/components/top_hits/hits/base.js

```diff
@@ -0,0 +1,321 @@
+import _ from 'underscore';
+import {
+    MAX_WORDS_SHORT,
+    SCIENTIFIC_NAME,
+    TERM_NAME,
+    WHITESPACE,
+} from '../constants';
+import { fieldToComponent } from './parts';
+
+
+/**
+* Shortens a sentence to maxWords.
+* @param {string} value - Sentence to trim.
+* @param {number} maxWords - Max number of words to return.
+* @return {string} Trimmed sentence.
+*/
+export const maybeTrimByWord = (value, maxWords) => {
+    const splitValue = value.split(WHITESPACE);
+    if (splitValue.length > maxWords) {
+        return `${splitValue.slice(0, maxWords).join(WHITESPACE)}...`;
+    }
+    return value;
+};
+
+
+/**
+* Some fields are lists and should be converted to strings.
+* @param {string|array} value - String or array to convert to string.
+* @param {string} sep - Token to join array on.
+* @return {string} Original string or joined array.
+*/
+export const maybeConvertArrayToString = (value, sep = ', ') => {
+    if (Array.isArray(value)) {
+        return value.join(sep);
+    }
+    return value;
+};
+
+
+/**
+* Helper function to filter arrays of undefined values and only keep
+* unique values.
+* @param {array} values - Array to filter and make unique.
+* @return {array} Unique array with only truthy values.
+*/
+export const uniqueAndDefinedValues = (values) => (
+    _.uniq(
+        values.filter(
+            (value) => Boolean(value)
+        )
+    )
```

Comment:
Underscore.js has a _.compact() function that filters out falsy entries in an array, though it doesn’t really cut things down much over what you have.

Reply:
Cool, updated to use.